Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/sites/chan55/Chan55.java

```diff
@@ -0,0 +1,308 @@
+    private static final String BASE_URL = "https://55chan.org";
+    private final Random secureRandom = new SecureRandom();
+
+    @Override
+    public void setup() {
+        setName("55chan");
+        setIcon(SiteIcon.fromFavicon(HttpUrl.parse("https://55chan.org/static/favicon.ico?v=2")));
+        setBoardsType(BoardsType.STATIC);
+
+        setResolvable(URL_HANDLER);
+
+        setConfig(new CommonConfig() {
+            @Override
+            public boolean feature(Feature feature) {
+                return feature == Feature.POSTING || feature == Feature.POST_DELETE;
+            }
+        });
+
+        setEndpoints(new VichanEndpoints(this,
+                BASE_URL,
+                BASE_URL) {
+            @Override
+            public HttpUrl imageUrl(Post.Builder post, Map<String, String> arg) {
+                return root.builder().s(post.board.code).s("src").s(arg.get("tim") + "." + arg.get("ext")).url();
+            }
+
+            @Override
+            public HttpUrl thumbnailUrl(Post.Builder post, boolean spoiler, Map<String, String> arg) {
+                String ext;
+                switch (arg.get("ext")) {
+                    case "jpeg":
+                    case "jpg":
+                    case "png":
+                    case "gif":
+                        ext = arg.get("ext");
+                        break;
+                    default:
+                        ext = "jpg";
+                        break;
+                }
+
+                return root.builder().s(post.board.code).s("thumb").s(arg.get("tim") + "." + ext).url();
+            }
+
+            @Override
+            public HttpUrl boards() {
+                return root.builder().s("boards.json").url();
+            }
+
+            @Override
+            public HttpUrl reply(Loadable loadable) {
+                return sys.builder().s("altpost.php").url();
+            }
+        });
+
+        setActions(new VichanActions(this) {
+            @Override
+            public void setupPost(Reply reply, MultipartHttpCall call) {
+                super.setupPost(reply, call);
+
+                if (reply.loadable.isThreadMode()) {
+                    // "thread" is already added in VichanActions.
+                    call.parameter("post", "Responder");
+                } else {
+                    call.parameter("post", "New Thread");
+                    call.parameter("page", "1");
+                }
+
+                call.parameter("json_response", "1");
+            }
+
+            @Override
+            public void handlePost(ReplyResponse replyResponse, Response response, String result) {
+                Matcher auth = Pattern.compile("dose").matcher(result);
+                Matcher banned = Pattern.compile("banned").matcher(result);
+                Matcher err = Pattern.compile("error").matcher(result);
+                Matcher postMatcher = Pattern.compile("\\.html#\\d+").matcher(result);
```

Comment:
All of these matchers should be made as static class variables to avoid their creation every time this method gets called.